aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SyncProcessor.java

```diff
@@ -154,15 +153,9 @@ private Completable createHydrationTask(Class<? extends Model> modelClass) {
                                         metricsAccumulator.toModelSyncedEvent(syncType).toHubEvent());
                 }));
             })
-            .doOnError(failureToSync -> {
-                LOG.warn("Initial cloud sync failed.", failureToSync);
-                DataStoreErrorHandler dataStoreErrorHandler =
-                    dataStoreConfigurationProvider.getConfiguration().getErrorHandler();
-                dataStoreErrorHandler.accept(new DataStoreException(
-                    "Initial cloud sync failed.", failureToSync,
-                    "Check your internet connection."
-                ));
-            })
+            .doOnError(failureToSync ->
+                LOG.warn("Initial cloud sync failed.", failureToSync)
```

Comment:
Why would we not want to call the error handler if something happens during the full/delta sync?

Reply:
For now, Amplify only handles errors from mutation, and anything during query/subscription is purposefully ignored (across all platforms)

Reply:
My understanding is that this is the long term plan as well.  The general philosophy is that we should not make the client deal with errors if at all possible.  The only thing the client likely cares about is whether datastore is in sync, or not, which they can get from the [Hub's networkStatus event](https://docs.amplify.aws/lib/datastore/datastore-events/q/platform/android).   So, for sync and subscription failures, we should just handle these internally, by retrying with exponential backoff (already exists for iOS/JS, and something we should try to add in our next sprint)

Mutations are a special case though.  If a mutation fails for an unrecoverable reason (e.g. user is `Unauthorized`), we don't want it keep it in the persistent mutation queue, since it will just continually fail over and over again and block the queue.  We also can't remove it though, because we don't want to delete customer data.  We can, however, delete it after giving it back to the customer via the error handler.